### PR TITLE
SI-7695 Macro debug output on -explaintypes

### PR DESCRIPTION
**Pull request by som-snytt**
It looks like checkConforms should wait for error before
using explainTypes, rather than turning on macro debug
trace if explaintypes is set.

In particular, macro debug output is Console.println
instead of reporter.echo or similar, so partest can't
test it.

### CODE REVIEW

File: src/compiler/scala/reflect/macros/compiler/Errors.scala [hunk 1]
```diff
@@ -70,9 +70,10 @@ trait Errors extends Traces {
       if (verbose) withTypesExplained(check(rtpe, atpe))
       else check(rtpe, atpe)
     if (!ok) {
-      if (!macroDebugVerbose)
-        explainTypes(rtpe, atpe)
-      compatibilityError("type mismatch for %s: %s does not conform to %s".format(slot, abbreviateCoreAliases(rtpe.toString), abbreviateCoreAliases(atpe.toString)))
+      if (!verbose) explainTypes(rtpe, atpe)
+      val rCA = abbreviateCoreAliases(rtpe.toString)
+      val aCA = abbreviateCoreAliases(atpe.toString)
```
Comment (xeno-by) [inline src/compiler/scala/reflect/macros/compiler/Errors.scala]: Not sure about the names.

File: src/compiler/scala/reflect/macros/compiler/Errors.scala [hunk 2]
```diff
@@ -53,7 +53,7 @@ trait Errors extends Traces {
   // not exactly an error generator, but very related
   // and I dearly wanted to push it away from Macros.scala
   private def checkConforms(slot: String, rtpe: Type, atpe: Type) = {
-    val verbose = macroDebugVerbose || settings.explaintypes.value
+    val verbose = macroDebugVerbose
```
Comment (xeno-by) [inline src/compiler/scala/reflect/macros/compiler/Errors.scala:56]: I think that shutting up when everything's okay is the right thing to do.

**Comment by gkossakowski** [comment]
@som-snytt, @xeno-by: what's the status of this PR?

**Comment by xeno-by** [comment]
lgtm

**Pull request merged**